### PR TITLE
feat: add registration page

### LINKED ISSUE #50: Registration Page
# TBD requirements

- Resumes upload
- User email
- Name
- School
- Program
- Year
- Agree terms of use and code of conduct
- Analytics (where did you hear about CUSEC?)

Dispatch an email to the user upon successful registration through Firebase? (With Hopin registration link)

### PR DESCRIPTION
**Pull request by sttic**
Adds a /register page that is an embedded full-page Typeform form.

Note:

- The linked Typeform is still being finalized. The link stays the same after updates, so it will still reflect the most recent version after being merged.
- The pro plan for Typeform is being used so it's up to 5000 responses per month

Resolves #50

### CODE REVIEW

**Comment by gordonpn** [comment]
Oh, we're using Typeform after all?

With GitHub student pack, we have access to this:

image

Are we using the free tier and what are the limitations? It won't let me see the differences without making an account lol

image

**Comment by sttic** [comment]
> Oh, we're using Typeform after all?

The integration looks ok and their forms look nice. It'd save us a lot of time which is nice :joy: 

I didn't know we pro Typeform access with the GitHub student pack, good find!

> Are we using the free tier and what are the limitations?

I tried free tier and it had a limit of responses (something like 100 or so) and some restrictions on publishing it. The pro plan looks like it might be enough!

**Comment by gordonpn** [comment]
> The integration looks ok and their forms look nice. It'd save us a lot of time which is nice 😂

I agree, I believe I've signed up to Hackathons and other conferences with Typeform as well!

The pro plan should be plenty then! My concern was mostly for the first month when the registration is announced, we may get more than 100 sign-ups in that month.

File: pages/register.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      </Head>
+
+      <iframe
+        id="typeform-full"
+        width="100%"
+        height="100%"
+        frameBorder="0"
+        allow="camera; microphone; autoplay; encrypted-media;"
+        src="https://eli030.typeform.com/to/FFoe9hQ9"
+      ></iframe>
```
Comment (gordonpn) [inline pages/register.tsx:22]: Kind of surprised ESLint didn't say anything about using a self-closing tag here (and below script) 😂

**Review by gordonpn — APPROVED** [review]
Thank you, Tommy!

File: pages/register.tsx (hunk 1 above)
Comment (sttic) [reply pages/register.tsx:22]: Yeah 😂 I pretty much pasted the exact embed code typeform provided and it didn't complain

**Pull request merged**